plugins/capture.rb

```diff
@@ -0,0 +1,612 @@
+      end
+
+      if @active_loggers.key?(session)
+        logger = @active_loggers[session]
+        logger.close
+      end
+
+      # Start afresh
+      @active_job_ids[session] = []
+      @active_loggers.delete(session)
+
+      transform_params(config)
+      validate_params(config)
+
+      modules = {
+        # Capturing
+        'DRDA' => 'auxiliary/server/capture/drda',
+        'FTP' => 'auxiliary/server/capture/ftp',
+        'IMAP' => 'auxiliary/server/capture/imap',
+        'MSSQL' => 'auxiliary/server/capture/mssql',
+        'MySQL' => 'auxiliary/server/capture/mysql',
+        'POP3' => 'auxiliary/server/capture/pop3',
+        'Postgres' => 'auxiliary/server/capture/postgresql',
+        'PrintJob' => 'auxiliary/server/capture/printjob_capture',
+        'SIP' => 'auxiliary/server/capture/sip',
+        'SMB' => 'auxiliary/server/capture/smb',
+        'SMTP' => 'auxiliary/server/capture/smtp',
+        'Telnet' => 'auxiliary/server/capture/telnet',
+        'VNC' => 'auxiliary/server/capture/vnc',
+
+        # SSL versions
+        'FTPS' => 'auxiliary/server/capture/ftp',
+        'IMAPS' => 'auxiliary/server/capture/imap',
+        'POP3S' => 'auxiliary/server/capture/pop3',
+        'SMTPS' => 'auxiliary/server/capture/smtp',
+
+        # Poisoning
+        #'DNS' => 'auxiliary/spoof/dns/native_spoofer',
+        'NBNS' => 'auxiliary/spoof/nbns/nbns_response',
+        'LLMNR' => 'auxiliary/spoof/llmnr/llmnr_response',
+        'mDNS' => 'auxiliary/spoof/mdns/mdns_response',
+        #'WPAD' => 'auxiliary/server/wpad',
+      }
+
+      encrypted = ['HTTPS_NTLM','HTTPS_Basic','FTPS','IMAPS','POP3S','SMTPS']
+
+      if config[:http_basic]
+        modules['HTTP'] = 'auxiliary/server/capture/http_basic'
+        modules['HTTPS'] = 'auxiliary/server/capture/http_basic'
+      else
+        modules['HTTP'] = 'auxiliary/server/capture/http_ntlm'
+        modules['HTTPS'] = 'auxiliary/server/capture/http_ntlm'
+      end
+
+      modules_to_run = []
+      logfile = config[:logfile]
+      print_line("Logging results to #{logfile}")
+      logdir = ::File.dirname(logfile)
+      FileUtils.mkdir_p(logdir)
+      hashdir = config[:hashdir]
+      print_line("Hash results stored in #{hashdir}")
+      FileUtils.mkdir_p(hashdir)
+
+
+      if config[:stdout]
+        logger = Rex::Ui::Text::Output::Tee.new(logfile)
+      else
+        logger = Rex::Ui::Text::Output::File.new(logfile, mode='ab')
+      end
+
+      @active_loggers[session] = logger
+
+      modules.each do |svc, module_name|
+        unless config[:services][svc]
+          # This service turned off in config
+          next
+        end
+        # Special case for two variants of HTTP
+        if svc.start_with?('HTTP')
+          if config[:http_basic]
+            svc += '_Basic'
+          else
+            svc += '_NTLM'
+          end
+        end
+
+        mod = framework.modules.create(module_name)
+        # Bail if we couldn't
+        unless mod
```

Comment:
There's a pattern for extracting module load errors that we follow over here https://github.com/rapid7/metasploit-framework/blob/d6ff12f23b289539b4126d1c67197251e109785f/lib/msf/ui/console/command_dispatcher/modules.rb#L718-L740

Reply:
Thanks. I guess in that case, it's doing some searching to try to figure out what the user wants; in this case, we're explicitly specifying the exact module, and don't want it to go searching. So I'd imagine that most of that we don't want to use; but I can use the `load_error_by_name` method to give a better error?